source/application/AMPIC887DriverServer/AMPIC887AxisMap.h

```diff
@@ -0,0 +1,473 @@
+		foreach(AMGCS2::Axis currentAxis, keys) {
+			if(contains(currentAxis)) {
+				returnValues.insert(currentAxis, value(currentAxis));
+			}
+		}
+
+		return returnValues;
+	}
+};
+
+template <>
+class AMPIC887AxisMap<AMGCS2::PositionUnits> : public QMap<AMGCS2::Axis, AMGCS2::PositionUnits>
+{
+public:
+	/*!
+	  * Creates an instance of an axis map, which is empty.
+	  */
+	AMPIC887AxisMap()
+		: QMap<AMGCS2::Axis, AMGCS2::PositionUnits>()
+	{
+	}
+
+	/*!
+	  * Returns a collection of the axes which this map contains.
+	  */
+	AMPIC887AxisCollection axes() const
+	{
+		return AMPIC887AxisCollection(keys());
+	}
+
+	/*!
+	  * Outputs the map values as a human readable string, if it can parse the
+	  * type into a string, otherwise returns the empty string.
+	  */
+	QString toString(const QString& header = "Data") const
+	{
+		QString returnString = QString("Axis\t\t%1\n").arg(header);
+
+		QList<AMGCS2::Axis> axes =  keys();
+		foreach(AMGCS2::Axis currentAxis, axes) {
+
+			returnString.append(QString("%1\t\t%2\n")
+								.arg(AMGCS2Support::axisToCharacter(currentAxis))
+								.arg(AMGCS2Support::positionUnitsToString( value(currentAxis))));
+		}
+
+		return returnString.trimmed();
+	}
+
+	/*!
+	  * Whether this map contains an entry for the UnknownAxis.
+	  */
+	bool containsUnknownAxis() const
+	{
+		return  contains(AMGCS2::UnknownAxis);
+	}
+
+	/*!
+	  * Whether this map contains entries for all valid axes
+	  */
+	bool containsAllAxes() const
+	{
+		return !containsUnknownAxis() &&
+				this->count() == AXIS_COUNT;
+	}
+
+	/*!
+	  * Whether this map contains an entry for a rotational axis
+	  */
+	bool containsRotationalAxes() const
+	{
+		QList<AMGCS2::Axis> axes = keys();
+		foreach(AMGCS2::Axis currentAxis, axes) {
+
+			if(currentAxis == AMGCS2::UAxis ||
+					currentAxis == AMGCS2::VAxis ||
+					currentAxis == AMGCS2::WAxis) {
+
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+
+	/*!
+	  * Whether this map contains an entry for a linear axis
+	  */
+	bool containsLinearAxis() const
+	{
+		QList<AMGCS2::Axis> axes = keys();
+		foreach(AMGCS2::Axis currentAxis, axes) {
+
+			if(currentAxis == AMGCS2::XAxis ||
+					currentAxis == AMGCS2::YAxis ||
+					currentAxis == AMGCS2::ZAxis) {
+
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/*!
+	  * A map of the provided axes to their values stored within this map. If
+	  * a key contained within the provided collection is not contained within
+	  * this map it will not be included in the returned map.
+	  * \param keys ~ A collection of axes whose values within this map will
+	  * be returned.
+	  */
+	AMPIC887AxisMap<AMGCS2::PositionUnits> matchingValues(const AMPIC887AxisCollection& keys) const
+	{
+		AMPIC887AxisMap<AMGCS2::PositionUnits> returnValues;
+		foreach(AMGCS2::Axis currentAxis, keys) {
+			if( contains(currentAxis)) {
+				returnValues.insert(currentAxis,  value(currentAxis));
+			}
+		}
+
+		return returnValues;
+	}
+};
+
+#endif // AMPIC887AXISMAP_H
```

Comment:
Looks like https://patch-diff.githubusercontent.com/raw/acquaman/acquaman/pull/2006.patch can show all the diffs, but we can't comment on it.